tlsn/examples/simple_prover.rs

```diff
@@ -0,0 +1,304 @@
+    let (mut request_sender, connection) =
+        hyper::client::conn::handshake(mpc_tls_connection.compat())
+            .await
+            .unwrap();
+
+    // Spawn the HTTP task to be run concurrently
+    let connection_task = tokio::spawn(connection.without_shutdown());
+
+    // Build a simple HTTP request with common headers
+    let request = Request::builder()
+        .uri(format!("https://{SERVER_DOMAIN}"))
+        .header("Host", SERVER_DOMAIN)
+        .header("Accept", "*/*")
+        // Using "identity" instructs the Server not to use compression for its HTTP response.
+        // TLSNotary does not support compressed responses.
+        .header("Accept-Encoding", "identity")
+        .header("Connection", "close")
+        .header("User-Agent", USER_AGENT)
+        .body(Body::empty())
+        .unwrap();
+
+    debug!("Sending request");
+
+    // Send the request to the Server and get a response via the MPC TLS connection
+    let response = request_sender.send_request(request).await.unwrap();
+
+    debug!("Sent request, got response {:?}", response);
+
+    assert!(response.status() == StatusCode::OK);
+
+    debug!("Response OK");
+
+    // Close the connection to the server
+    let mut client_socket = connection_task.await.unwrap().unwrap().io.into_inner();
+    client_socket.close().await.unwrap();
+
+    // The Prover task should be done now, so we can grab the Prover.
+    let mut prover = prover_task.await.unwrap().unwrap();
+
+    // Prepare for selective disclosure.
+
+    // Identify the ranges in the outbound data which contain data which we want to disclose
+    let (public_ranges, _) = find_ranges(
+        prover.sent_transcript().data(),
+        &[
+            // Redact the value of the "User-Agent" header. It will NOT be disclosed.
+            USER_AGENT.as_bytes(),
+        ],
+    );
+
+    // Commit to each range of the outbound data which we want to disclose
+    for range in public_ranges.iter() {
+        prover.add_commitment_sent(range.clone()).unwrap();
+    }
+
+    // Commit to all inbound data in one shot, as we don't need to redact anything in it
+    let recv_len = prover.recv_transcript().data().len();
+    prover.add_commitment_recv(0..recv_len as u32).unwrap();
+
+    // Finalize, returning the notarized session
+    let notarized_session = prover.finalize().await.unwrap();
+
+    debug!("Notarization complete!");
+
+    // Create a proof for all committed data in this session
+    let session_proof = notarized_session.session_proof();
+    let ids = (0..notarized_session.data().commitments().len()).collect();
+    let substrings_proof = notarized_session.generate_substring_proof(ids).unwrap();
+
+    // Write the proof to a file in the format expected by `simple_verifier.rs`
+    let mut file = tokio::fs::File::create("proof.json").await.unwrap();
+    file.write_all(
+        serde_json::to_string_pretty(&(&session_proof, &substrings_proof, &SERVER_DOMAIN))
```

Comment:
since the slowdown is seemingly insignificant but there is an added benefit of someone curious poking into the .json file to see what's inside, i prefer to keep it as json